src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2492,4 +2522,68 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     {
         _playWarningBell->Run();
     }
+
+    Windows::UI::Xaml::DependencyProperty VisualBellLight::m_isTargetProperty =
+        Windows::UI::Xaml::DependencyProperty::RegisterAttached(
+            L"IsTarget",
+            winrt::xaml_typename<bool>(),
+            winrt::xaml_typename<winrt::Microsoft::Terminal::Control::VisualBellLight>(),
+            Windows::UI::Xaml::PropertyMetadata{ winrt::box_value(false), Windows::UI::Xaml::PropertyChangedCallback{ &VisualBellLight::OnIsTargetChanged } });
+
+    void VisualBellLight::OnConnected(Windows::UI::Xaml::UIElement const& /* newElement */)
+    {
+        if (!CompositionLight())
+        {
+            // OnConnected is called when the first target UIElement is shown on the screen. This enables delaying composition object creation until it's actually necessary.
+            auto spotLight2{ Windows::UI::Xaml::Window::Current().Compositor().CreateAmbientLight() };
+            spotLight2.Color(Windows::UI::Colors::WhiteSmoke());
+            spotLight2.Intensity(static_cast<float>(1.5));
+            CompositionLight(spotLight2);
+        }
+    }
+
+    void VisualBellLight::OnDisconnected(Windows::UI::Xaml::UIElement const& /* oldElement */)
+    {
+        // OnDisconnected is called when there are no more target UIElements on the screen.
+        // Dispose of composition resources when no longer in use.
+        if (CompositionLight())
+        {
+            CompositionLight(nullptr);
+        }
+    }
+
+    winrt::hstring VisualBellLight::GetId()
+    {
+        return VisualBellLight::GetIdStatic();
+    }
+
+    void VisualBellLight::OnIsTargetChanged(Windows::UI::Xaml::DependencyObject const& d, Windows::UI::Xaml::DependencyPropertyChangedEventArgs const& e)
+    {
+        auto uielem{ d.try_as<Windows::UI::Xaml::UIElement>() };
+        auto brush{ d.try_as<Windows::UI::Xaml::Media::Brush>() };
+
+        auto isAdding = winrt::unbox_value<bool>(e.NewValue());
+        if (isAdding)
+        {
+            if (uielem)
+            {
+                Windows::UI::Xaml::Media::XamlLight::AddTargetElement(VisualBellLight::GetIdStatic(), uielem);
+            }
+            else if (brush)
+            {
+                Windows::UI::Xaml::Media::XamlLight::AddTargetBrush(VisualBellLight::GetIdStatic(), brush);
+            }
+        }
+        else
+        {
+            if (uielem)
+            {
+                Windows::UI::Xaml::Media::XamlLight::RemoveTargetElement(VisualBellLight::GetIdStatic(), uielem);
+            }
+            else if (brush)
+            {
+                Windows::UI::Xaml::Media::XamlLight::RemoveTargetBrush(VisualBellLight::GetIdStatic(), brush);
+            }
+        }
+    }

```

Comment:
Should these be in their own file? Also, if we add more lights in the future we probably want a macro to get all this set up

Reply:
> Should these be in their own file?
I vote yes. TermControl is already pretty big haha.

> Also, if we add more lights in the future we probably want a macro to get all this set up
If you add this into its own file, adding the macro should work really nicely. That way the macro is only exposed to the relevant part of the code ;)